From: Mauro Carvalho Chehab <mchehab@redhat.com>
Date: Thu, 12 May 2011 01:36:47 +0000 (-0300)
Subject: [media] Use a more consistent value for RC repeat period
X-Git-Url: http://git.linuxtv.org/media_tree.git?a=commitdiff_plain;h=ca540c8b66874aaf704c64d43d33e39369c428fb

[media] Use a more consistent value for RC repeat period

The default REP_PERIOD is 33 ms. This doesn't make sense for IR's,
as, in general, an IR repeat scancode is provided at every 110/115ms,
depending on the RC protocol. So, increase its default, to do a
better job avoiding ghost repeat events.

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
Acked-by: Jarod Wilson <jarod@redhat.com>
---

diff --git a/drivers/media/rc/rc-main.c b/drivers/media/rc/rc-main.c
index 0d4fcd9..f57cd56 100644
--- a/drivers/media/rc/rc-main.c
+++ b/drivers/media/rc/rc-main.c
@@ -1078,6 +1078,13 @@ int rc_register_device(struct rc_dev *dev)
 	 */
 	dev->input_dev->rep[REP_DELAY] = 500;
 
+	/*
+	 * As a repeat event on protocols like RC-5 and NEC take as long as
+	 * 110/114ms, using 33ms as a repeat period is not the right thing
+	 * to do.
+	 */
+	dev->input_dev->rep[REP_PERIOD] = 125;
+
 	path = kobject_get_path(&dev->dev.kobj, GFP_KERNEL);
 	printk(KERN_INFO "%s: %s as %s\n",
 		dev_name(&dev->dev),
